### PR TITLE
replace manager exceptions with debug.logerror and return

### PR DESCRIPTION
**Pull request by davidkline-ms**
The manager should not throw exceptions. They are expensive and handled automatically, so they aren't much more useful than a simple (and quicker) debug message

Changes
---
* The manager no longer throws
* Error debug log messages are displayed
* Safe (null, empty collections, etc) values are returned where an exception would have been thrown

### CODE REVIEW

File: Assets/MixedRealityToolkit/_Core/Managers/MixedRealityManager.cs [hunk 1]
```diff
@@ -786,7 +848,11 @@ public void EnableManager(Type type, string managerName)
         /// <returns>An array of Managers that meet the search criteria</returns>
         public IEnumerable<IMixedRealityManager> GetManagers(Type type)
         {
-            if (type == null) { throw new ArgumentNullException(nameof(type)); }
+            if (type == null)
+            {
+                Debug.LogWarning("Unable to get managers with a type of null.");
+                return new List<IMixedRealityManager>();
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit/_Core/Managers/MixedRealityManager.cs]: shouldn't this return null?

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Managers/MixedRealityManager.cs]: (same question for similar returns)

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/_Core/Managers/MixedRealityManager.cs]: if it returns null, then code that expects to be able to enumerate over the collection, for example

foreach (IMixedRealityManager mgr in MixedRealityManager,Instance.GetManagers(type) 

will throw an exception, or have to be rewritten to cache the result of GetManagers before the loop.

**Comment by davidkline-ms** [comment]
trying to get checks to resolve....

**Comment by StephenHodgson** [comment]
Yup. Likely Github is trying to recover still.

**Pull request merged**